src/dsvpn.c

```diff
@@ -752,9 +818,22 @@ client_reconnect(Context* context)
     return -1;
 }
 
+static int
+exit_handler(Context* context) {
+    if(del_firewall_rules(context) == -1) {
+        exit(-1);
```

Comment:
This is the return code of the command, so it can't be negative. It should probably be `1` if `del_firewall_rules()` failed, and `0` otherwise.

Reply:
Thanks I was really not sure about this. BTW, why don't you use `EXIT_FAILURE` and `EXIT_SUCCESS`?

Reply:
You can use `EXIT_FAILURE` and `EXIT_SUCCESS`, but that made only a difference 30 years ago on VMS.

Reply:
Ok :+1: